### ISSUE TITLE
CImageGrabber_OpenCV does not work with OpenCV 3.4.3 and Windows 10

### ISSUE THREAD

**Issue by Wehrdo** [MRPT/mrpt#852]
Attempting to open a webcam using [CCameraSensor](http://mrpt.ual.es/reference/devel/classmrpt_1_1hwdrivers_1_1_c_camera_sensor.html) with the suggested code sample will connect to the camera, but fail to retrieve frames.

```cpp
CCameraSensor cam;
const string cam_config_str =
    "[CONFIG]\n"
    "grabber_type=opencv\n";
CConfigFileMemory cam_config(cam_config_str);
cam.loadConfig(cam_config, "CONFIG");
cam.initialize();
auto obs = cam.getNextFrame();
```

It looks like two things have conspired to cause failure.

1. CImageGrabber_OpenCV uses the "legacy" C API for video capture, which no longer supports DirectShow --- instantiating a camera with the above code prints this out in the console:
`[ WARN:0] cvCreateFileCaptureWithPreference: backend DSHOW doesn't support legacy API anymore.`
(OpenCV cycles through backends and attempts them [[source]](https://github.com/opencv/opencv/blob/e6dd9a78aee8ccd217114f2b530e9772cdb6eb81/modules/videoio/src/videoio_c.cpp#L37))

2. Not sure what the root cause is, but the Video for Windows (VFW) backend no longer works with OpenCV on recent versions of Windows 10. I'm apparently not the only one with [this issue](https://github.com/opencv/opencv/issues/11810).

It looks like the solution would be updating CImageGrabber_OpenCV to use the `cv::VideoCapture` C++ API, since it can use DSHOW on Windows.

To verify these conclusions, connecting to my camera with `cv::VideoCapture(0)` on OpenCV 3.4.3 works fine, but with `cv::VideoCapture(0 + CAP_VFW)` results in the same behavior as with MRPT: Camera successfully opens, but frames fail to retrieve. Also, stepping through the MRPT code, I can see that the default `CAMERA_CV_AUTODETECT` resolves to using the VFW backend.

**Comment by jlblancoc** [comment]
Thanks for nailing it down! We'll fix it...

**Comment by jlblancoc** [comment]
@Wehrdo let me know if the code in the branch [opencv-fixes](https://github.com/MRPT/mrpt/tree/opencv-fixes) does work now... 

For some reason I upgraded to opencv 3.4.3 and now I have some problems due to (apparently unrelated) missing DLLS (MF.DLL, etc.) 😞

**Comment by jlblancoc** [comment]
Well, the current implementation doesn't work yet... it seems I'm unable to properly convert from cv::Mat to IplImage without causing a crash, sigh... Will continue with this another day, but any helping hand will be more than welcome! ;-)

**Comment by jlblancoc** [comment]
> Converting the IplImage to cv::Img is one of the oldest tasks on our issue tracker 

Yep... still, it's frustrating that the one causing problems is the other way around! I modernized the capture code to use the "modern" (8 years old now?) cv::Mat, but then when storing it into mrpt::img::CImage, I convert it into IplImage. 
In theory, IplImage offers a ctor to do the job... but I've tried it with IplImage in the stack, in the heap, and all seems to work but opencv eventually crashes while saving the CImage to disk, or trying to display it to a GUI... (sigh)

**Comment by jolting** [comment]
There is still a mismatch between `cv::Img` and` IplImage`. The best example of this is `imageDataOrigin`. Arguably removing this in `cv::Img` was a good choice, but I can see how some APIs wouldn't be happy converting back/forth.

**Comment by jlblancoc** [comment]
This issue should be fixed in master after fe0d0ab652e65803854683f661a8dde27ef93e48. 
Reopen if it's not! 
Cheers
